### DISCUSSION TITLE
How to add and delete from a set type?

### DISCUSSION THREAD

**Discussion by meightythree** [tywalch/electrodb#390] [Q&A] [answered]
```
import { Entity, Service } from 'electrodb';

export const ConnectionEntity = new Entity(
    {
        model: {
            entity: 'connections',
            version: '1',
            service: 'myApp',
        },
        attributes: {
            userId: {
                type: 'string',
                required: true,
                readOnly: true,
            },
            connectionIds: {
                type: 'set',
                items: 'string',
                default: () => [],
                required: true,
            },
        },
        indexes: {
            session: {
                pk: {
                    field: 'userId',
                    composite: ['userId'],
                },
            },
        },
    },
    {
        table: 'connections',
    },
);
```

Is it possible to directly add/delete from the `connectionIds` set without getting the entity?

**Comment by raryanpur** [comment, marked as answer]
Sure, you should be able to use the `add` and `delete` operations on the `patch` and `update` mutations (docs [here](https://electrodb.dev/en/mutations/patch/#add) and [here](https://electrodb.dev/en/mutations/patch/#delete)). These resolve to using the `UPDATE` expression in Dynamo under the hood, so don't require you to read-modify-write in client code or use read capacity units.

For example,

`await ConnectionEntity.patch({userId}).delete({connectionIds: [<ids_to_remove>]})`
